### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by dreyhiflden**

### CODE REVIEW

**Comment by dreyhiflden** [comment]
Only first 2 tasks complete

File: ball.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  if (event.clientX < posLeft + (borderWidth + ballRadius) - scrollX) {
+    leftCoordinate = borderWidth + ballRadius;
+  }
+
+  if (event.clientX > posLeft + cords.width - (borderWidth + ballRadius) - scrollX) {
+    leftCoordinate = cords.width - (borderWidth + ballRadius);
+  }
+
+  ball.style.cssText = `top:${topCoordinate}px;
```
Comment (carrot-e) [inline ball.js]:
```
top: ${topCoordinate}px;
...
left: ${leftCoordinate}px;
```

File: ball.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+   margin-top: -${ballRadius + borderWidth}px;
+   margin-left: -${ballRadius + borderWidth}px;
+   left:${leftCoordinate}px`;
+}
```
Comment (carrot-e) [inline ball.js]: `;`

File: carousel.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  </div>
+
+  <script>
+    const elem = document.querySelector('.carousel'),
+      btns = document.querySelectorAll('button');
+
+    [...btns].map(el => el.addEventListener('click', move));
+
+    function move() {
+      console.log(elem.clientWidth);
```
Comment (carrot-e) [inline carousel.html]: Debugging code in final version?

File: carousel.html [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+      console.log(elem.scrollWidth);
+      this.className === 'left-arrow' ? elem.scrollLeft -= elem.clientWidth : elem.scrollLeft += elem.clientWidth
```
Comment (carrot-e) [inline carousel.html]: `;`

File: index.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Tasks list</title>
+</head>
+<body>
+  <ul>
+    <li><a href="carousel.html">Carousel</a></li>
+    <li><a href="move-ball-field.html.html">Move ball</a></li>
```
Comment (carrot-e) [inline index.html]: Check the link

File: index.html [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  </ul>
+</body>
+</html>
```
Comment (carrot-e) [inline index.html]: :scream:

File: move-ball-field.html [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    body {
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      flex-direction: column;
+      height: 800px;
+      width: 2000px;
+    }
+
+    #field {
```
Comment (carrot-e) [inline move-ball-field.html]: ID selector?

File: carousel.html [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  <meta charset="utf-8">
+  <link rel="stylesheet" href="style.css">
+</head>
+<body>
+
+  <div class="container">
+    <button class="left-arrow">⇦</button>
+
+    <ul class="carousel">
+      <li class="carousel__item"><img src="img/1.png" alt="image"></li>
```
Comment (carrot-e) [inline carousel.html]: screenshot from 2019-03-03 13-40-57

File: move-ball-field.html [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+      left: 0;
+      position: absolute;
+    }
+  </style>
+</head>
+
+<body>
+
+  <div id="field">
+    <img src="img/ball.svg" id="ball">
```
Comment (carrot-e) [inline move-ball-field.html]: screenshot from 2019-03-03 13-40-39

File: style.css [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+  width: 450px;
+  overflow-x: hidden;
+  background: #fff;
+  padding: 0;
+  align-items: center;
+  scroll-behavior: smooth;
+}
+
+.carousel__item {
+  padding: 0 10px 0 10px;
```
Comment (carrot-e) [inline style.css]: Can be shortened.

File: ball.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  if (event.clientX < posLeft + (borderWidth + ballRadius) - scrollX) {
+    leftCoordinate = borderWidth + ballRadius;
+  }
+
+  if (event.clientX > posLeft + cords.width - (borderWidth + ballRadius) - scrollX) {
+    leftCoordinate = cords.width - (borderWidth + ballRadius);
+  }
+
```
Comment (carrot-e) [inline ball.js]: `posTop - window.scrollY + borderWidth + ballRadius`
`(posTop + cords.height - window.scrollY) - (borderWidth + ballRadius)`
`posLeft + (borderWidth + ballRadius) - scrollX`
`posLeft + cords.width - (borderWidth + ballRadius) - scrollX`

It takes time to understand what each expression means.

You can create helper functions for each case to speed up understanding:
```
const getMinY = () => posTop - window.scrollY + borderWidth + ballRadius;
...
if (event.clientY < getMinY()) {
  ...
}
```

Makes sense?

File: carousel.html [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+    </ul>
+
+    <button class="right-arrow">⇨</button>
+  </div>
+
+  <script>
+    const elem = document.querySelector('.carousel'),
+      btns = document.querySelectorAll('button');
+
+    [...btns].map(el => el.addEventListener('click', move));
```
Comment (carrot-e) [inline carousel.html]: How about this:
```
[...document.querySelectorAll('button')]
  .forEach(el => el.addEventListener('click', move));
```

File: ball.js (hunk 2 above)
Comment (dreyhiflden) [reply ball.js]: My IDE suggest me don't place semicolon at 34 line. 
screenshot from 2019-03-03 14-29-33

File: ball.js (hunk 1 above)
Comment (dreyhiflden) [reply ball.js]: Add missing spaces

File: carousel.html (hunk 3 above)
Comment (dreyhiflden) [reply carousel.html]: Delete debugging from final version

File: carousel.html (hunk 4 above)
Comment (dreyhiflden) [reply carousel.html]: Add missed semicolon

File: move-ball-field.html (hunk 9 above)
Comment (dreyhiflden) [reply move-ball-field.html]: Add missed images

File: ball.js (hunk 2 above)
Comment (carrot-e) [reply ball.js]: Whoops, that's right. Sorry, my mistake!

File: index.html (hunk 5 above)
Comment (dreyhiflden) [reply index.html]: Fixed link

File: style.css (hunk 10 above)
Comment (dreyhiflden) [reply style.css]: Done

File: ball.js (hunk 11 above)
Comment (dreyhiflden) [reply ball.js]: Agree, the code looks clearer with the helper functions

File: carousel.html (hunk 12 above)
Comment (dreyhiflden) [reply carousel.html]: Looks better, gonna use this, because we don't need additional variable who used only once.

**Pull request merged**